### PR TITLE
feat: use `@next/font` for font loading

### PR DESCRIPTION
**Pull request by G3root**
## What does this PR do?

this PR uses `@next/font` for loading fonts. which has some performance benefits.

before:
Screenshot 2023-01-09 17:11:38

after:
Screenshot 2023-01-09 17:06:38

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/pages/_app.tsx [hunk 1]
```diff
@@ -21,6 +23,13 @@ import I18nLanguageHandler from "@components/I18nLanguageHandler";
 
 import "../styles/globals.css";
 
+const interFont = Inter({ subsets: ["latin"], variable: "--font-inter", preload: true });
+const calFont = localFont({ src: "../fonts/CalSans-SemiBold.woff2", variable: "--font-cal", preload: true });
+
+if (typeof document !== "undefined") {
```
Comment (G3root) [inline apps/web/pages/_app.tsx:29]: a workaround for font not applied in modals and tooltips.

Comment (PeerRich) [reply apps/web/pages/_app.tsx:29]: how come it doesnt work in modals and tooltips?

Comment (G3root) [reply apps/web/pages/_app.tsx:29]: next font works by optimizing the font at build time and assigning a hash to it. in order to work we have assign the hash to the `css variable` we provided. if we follow the same step as the [docs](https://nextjs.org/docs/basic-features/font-optimization#with-tailwind-css), `css variable` will be asssigned inside a container with an id `__next`. whereas modals render outside this container, so it cannot inherit the assigned value. assigning the variable at body seems to fix it.

**Comment by PeerRich** [comment]
NIT: do we have benchmarks of before and after? curious

**Review by PeerRich — APPROVED** [review]
tested, works like a charm

**Review by PeerRich — DISMISSED** [review]
hold on, i think I saw a flashing font bug here: https://cal-git-fork-g3root-feature-next-font-optimzation-cal.vercel.app/pro


yup: https://www.loom.com/share/cdcaaab22aea4263b47081afd103f99a

**Comment by G3root** [comment]
> hold on, i think I saw a flashing font bug here: https://cal-git-fork-g3root-feature-next-font-optimzation-cal.vercel.app/pro
> 
> yup: https://www.loom.com/share/cdcaaab22aea4263b47081afd103f99a

i'll investigate it

File: apps/web/pages/_app.tsx (hunk 1 above)
Comment (emrysal) [reply apps/web/pages/_app.tsx]: Moved this into `:root` inspired by https://larsmagnus.co/blog/how-to-optimize-custom-fonts-with-next-font - this is also in the next docs (but maybe wasn't there a month ago)

**Review by emrysal — APPROVED** [review]
All good to go, tested this through locally and on preview, minor fixup with the inline style (Previously it would crash on SSR pages due to undefined document.body)

**Pull request merged**